gateway/client-switchover/README.md

```diff
@@ -0,0 +1,422 @@
+  --topic test-topic
+```
+
+- Consume from destination cluster mirror topic
+```
+kafka-console-consumer \
+  --bootstrap-server kafka-destination:9193 \
+  --consumer.config destination-cluster.config \
+  --topic test-topic \
+  --from-beginning \
+  --max-messages 2
+```
+
+## Migration Procedure: Blue/Green Deployment
+#### NOTE: Please modify the `bootstrap-server` config appropriately for all commands in this section.
+
+### Step 1: Pre-Flight Checks
+
+#### 1. Verify Cluster Link Status
+- Verify that replication lag is less than 100 messages per partition.
+```
+kafka-cluster-links \
+  --bootstrap-server kafka-destination:9193 \
+  --command-config destination-cluster.config \
+  --describe \
+  --link source-to-destination-link
+```
+
+#### 2. Check Offset Sync Status
+- Compare consumer group offsets between clusters. Validate that the offset difference corresponds to 30-60 seconds of messages.
+```bash
+# Get source offsets.
+kafka-consumer-groups --describe --all-groups \
+  --bootstrap-server kafka-source:9093 \
+  --command-config source-cluster.config > /tmp/source-offsets.txt
+
+# Get destination offsets.
+kafka-consumer-groups --describe --all-groups \
+  --bootstrap-server kafka-destination:9193 \
+  --command-config destination-cluster.config > /tmp/destination-offsets.txt
+  
+# Validate the difference.
+diff /tmp/source-offsets.txt /tmp/destination-offsets.txt
+```
+
+#### 3. Verify all topics are mirrored
+```bash
+# Get source kafka topics.
+kafka-topics --list --bootstrap-server kafka-source:9093  --command-config source-cluster.config | sort > /tmp/source-topics.txt
+
+# Get mirrored topics in the destination.
+kafka-mirrors \
+  --bootstrap-server kafka-destination:9193 \
+  --command-config destination-cluster.config \
+  --list \
+  --link source-to-destination-link | sort > /tmp/destination-topics.txt
+
+# Validate that any difference is expected (test or internal topics only).
+diff /tmp/source-topics.txt /tmp/destination-topics.txt
+```
+
+### Step 2: Traffic cutover to Green cluster
+
+#### 1. Patch loadbalancer to point to the Green deployment
+- Patch the loadbalancer to change the label selectors as well as the `targetPort` for the Kafka bootstrap listener.
+- Modify the `targetPort` in the below patch command to point to the port of the Green gateway route endpoint.
+```
+kubectl patch service confluent-gateway-switchover-lb -n confluent --type='json' -p='[
+  {"op": "replace", "path": "/spec/selector/app", "value": "confluent-gateway-green"},
+  {"op": "replace", "path": "/spec/ports/0/targetPort", "value": 9696}
+]'
+```
+
+####  2. Test message consumption from new Gateway setup [Loadbalancer pointing to Green deployment]
+- You should be able to consume the test messages that were produced in `test-topic` as part of the Cluster Linking verification step.
+```
+kafka-console-consumer \
+  --bootstrap-server gateway.example.com:9595 \
+  --consumer.config destination-cluster.config \
```

Comment:
```suggestion
  --consumer.config client.properties \
```
we should use same client properties file before/after switchover. Current command gives notion of client requiring change.